sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala

```diff
@@ -213,4 +213,56 @@ class DescribeTableSuite extends command.DescribeTableSuiteBase
       assert("""\d+\s+bytes,\s+4\s+rows""".r.matches(stats))
     }
   }
+
+  test("desc table constraints") {
+    withNamespaceAndTable("ns", "pk_table", nonPartitionCatalog) { tbl =>
+      withTable("fk_table") {
+        sql(
+          s"""
+             |CREATE TABLE fk_table (id INT PRIMARY KEY) USING parquet
+        """.stripMargin)
+        sql(
+          s"""
+             |CREATE TABLE $tbl (
+             |  id INT,
+             |  a INT,
+             |  b STRING,
+             |  c STRING,
+             |  PRIMARY KEY (id),
+             |  CONSTRAINT fk_a FOREIGN KEY (a) REFERENCES fk_table(id) RELY,
+             |  CONSTRAINT uk_b UNIQUE (b),
+             |  CONSTRAINT uk_a_c UNIQUE (a, c),
+             |  CONSTRAINT c1 CHECK (c IS NOT NULL),
+             |  CONSTRAINT c2 CHECK (id > 0)
+             |)
+             |$defaultUsing
+        """.stripMargin)
+
+        var expectedConstraintsDdl = Array(
+          "pk_table_pk,CONSTRAINT pk_table_pk PRIMARY KEY (id) NOT ENFORCED NORELY,",
+          "fk_a,CONSTRAINT fk_a FOREIGN KEY (a) REFERENCES fk_table (id) NOT ENFORCED RELY,",
+          "uk_b,CONSTRAINT uk_b UNIQUE (b) NOT ENFORCED NORELY,",
+          "uk_a_c,CONSTRAINT uk_a_c UNIQUE (a, c) NOT ENFORCED NORELY,",
+          "c1,CONSTRAINT c1 CHECK (c IS NOT NULL) ENFORCED NORELY,",
+          "c2,CONSTRAINT c2 CHECK (id > 0) ENFORCED NORELY,"
+        )
+        var descDdL = sql(s"DESCRIBE EXTENDED $tbl").collect().map(_.mkString(","))
```

Comment:
<img width="880" height="188" alt="image" src="https://github.com/user-attachments/assets/f4b73441-5ada-4815-bae9-bc7f8745a564" />
I realize that DDL contains the constraint name, which is quite duplicated. Also, we can skip showing "NOT ENFORCED"/"NORELY" if these clause are the default. 
I would suggest adding a `toDescription` method in `Constraint` for describing table purpose.